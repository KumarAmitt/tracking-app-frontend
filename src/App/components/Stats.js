import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { TARGET } from '../../constants';
import { getTotalPremium } from '../../store/slicers/deal';
import './style/Stats.css';

const Stats = ({ today }) => {
  const totalPremium = useSelector(getTotalPremium);

  return (
    <div className="stats">
      <div className="stats-item">
        <div>{today}</div>
        <div>Day</div>
      </div>
      <div className="stats-item">
        <div>{totalPremium}</div>
        <div>Achieved</div>
      </div>
      <div className="stats-item">
        <div>{TARGET > totalPremium ? TARGET - totalPremium : 0}</div>
        <div>Lag</div>
      </div>
      <div className="stats-item">
        <div>{TARGET}</div>
        <div>Target</div>
      </div>
    </div>
  );
};

Stats.propTypes = {
  today: PropTypes.number.isRequired,
};

export default Stats;
