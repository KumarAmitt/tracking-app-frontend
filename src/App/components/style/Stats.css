.stats {
    display: flex;
    justify-content: space-between;
    margin: 1em;
}

.stats-item {
    text-align: center;
}

.stats-item > *:first-child {
    font-weight: bold;
}

.stats-item > * + * {
    font-size: 0.7rem;
    text-transform: uppercase;
    margin-top: 0.4em;
    letter-spacing: 2px;
}