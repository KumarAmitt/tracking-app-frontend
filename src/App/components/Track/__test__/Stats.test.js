import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import Stats from '../Stats';

const store = configureAppStore();

describe('Stats', () => {
  it('should match the snapshot of Stats component', () => {
    const today = 300;
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Stats today={today} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
