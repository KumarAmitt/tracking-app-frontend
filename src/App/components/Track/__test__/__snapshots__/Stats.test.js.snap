// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats should match the snapshot of Stats component 1`] = `
<div
  className="stats"
>
  <div
    className="stats-item"
  >
    <div>
      300
    </div>
    <div>
      Day
    </div>
  </div>
  <div
    className="stats-item"
  >
    <div>
      0
    </div>
    <div>
      Achieved
    </div>
  </div>
  <div
    className="stats-item"
  >
    <div>
      700000
    </div>
    <div>
      Lag
    </div>
  </div>
  <div
    className="stats-item"
  >
    <div>
      700000
    </div>
    <div>
      Target
    </div>
  </div>
</div>
`;
